### PR TITLE
Converted getExecuteform to utilize twig template

### PR DESCRIPTION
**Pull request by britbarn**
Signed-off-by: Britt Barnes <britbarnes92@gmail.com>

### Description

Converted getExecuteForm to process through a twig template

### CODE REVIEW

File: libraries/classes/Database/Routines.php [hunk 1]
```diff
@@ -1471,65 +1441,52 @@ public function getExecuteForm(array $routine): string
...
-            $retval .= "<input type='hidden' name='execute_routine' value='true'>";
-            $retval .= "<input type='hidden' name='ajax_request' value='true'>";
         }
-        $retval .= "</form>\n\n";
-        $retval .= "<!-- END ROUTINE EXECUTE FORM -->\n\n";
 
-        return $retval;
+        return $this->template->render('database/routines/execute_form', [
+            'routine' => $routine,
+            'fromRoute' => Url::getFromRoute('/database/routines'),
```
Comment (williamdes) [inline libraries/classes/Database/Routines.php:1483]: This line and the next one are not needed or can be done in twig

File: templates/database/routines/execute_form.twig [hunk 2]
```diff
@@ -0,0 +1,65 @@
+<!-- START ROUTINE EXECUTE FORM -->
+{# {{ dump(getHiddenInputs) }} #}
```
Comment (williamdes) [inline templates/database/routines/execute_form.twig:2]:
```suggestion
```

File: libraries/classes/Database/Routines.php [hunk 3]
```diff
@@ -1410,56 +1410,26 @@ public function getExecuteForm(array $routine): string
...
-        }
-        $retval .= '<th>' . __('Value') . "</th>\n";
-        $retval .= "</tr>\n";
-        // Get a list of data types that are not yet supported.
         $no_support_types = Util::unsupportedDatatypes();
-        for ($i = 0; $i < $routine['item_num_params']; $i++) { // Each parameter
-            if ($routine['item_type'] === 'PROCEDURE'
-                && $routine['item_param_dir'][$i] === 'OUT'
+
+        $params = array();
```
Comment (williamdes) [inline libraries/classes/Database/Routines.php:1415]: please run `composer run phpcbf` and `composer run phpcs`

File: libraries/classes/Database/Routines.php [hunk 4]
```diff
@@ -1410,56 +1410,26 @@ public function getExecuteForm(array $routine): string
...
+        $params['no_support_types'] = $no_support_types;
+
+        for ($i = 0; $i < $routine['item_num_params']; $i++) {
+            if ($routine['item_type'] == 'PROCEDURE'
```
Comment (williamdes) [inline libraries/classes/Database/Routines.php:1419]: for things you are certain about you can use `===`

**Comment by britbarn** [comment]
@williamdes do you know what I need to do to fix the selenium test issues?

File: templates/database/routines/execute_form.twig [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+      </caption>
+      {% else %}
+      <legend> Routine parameters </legend>
+      <table class='rte_table'>
+        {% endif %}
+        <tr>
+          <th> Name </th>
+          <th> Type </th>
+          {% if cfg['ShowFunctionFields'] %}
+            <th>  Function </th>
```
Comment (williamdes) [inline templates/database/routines/execute_form.twig:21]:
```suggestion
            <th>{% trans 'Function' %}</th>
```

File: templates/database/routines/execute_form.twig [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+                <input class='{{  params[loop.index0]['class']  }}' type='text' name='params[{{  routine['item_param_name'][loop.index0]  }}]'>
+              {% endif %}
+            </td>
+          </tr>
+        {% endfor %}
+      </table>
+      {% if not ajax %}
+  </fieldset>
+  <fieldset class='tblFooters'>
+    <input type='submit' name='execute_routine' value='Go'>
```
Comment (williamdes) [inline templates/database/routines/execute_form.twig:57]:
```suggestion
    <input type='submit' name='execute_routine' value='{% trans 'Go' %}'>
```

File: templates/database/routines/execute_form.twig [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+  </fieldset>
+{% else %}
+  <input type='hidden' name='execute_routine' value='true'>
+  <input type='hidden' name='ajax_request' value='true'>
```
Comment (williamdes) [inline templates/database/routines/execute_form.twig:61]:
```suggestion
  <input type="hidden" name="execute_routine" value="true">
  <input type="hidden" name="ajax_request" value="true">
```

File: templates/database/routines/execute_form.twig [hunk 8]
```diff
@@ -0,0 +1,64 @@
+<!-- START ROUTINE EXECUTE FORM -->
+<form action="{{ url('/database/routines') }}" method="post" class="rte_form ajax" onsubmit="return false">
+  <input type='hidden' name='item_name' value='{{routine['item_name']}}'>
```
Comment (williamdes) [inline templates/database/routines/execute_form.twig:3]: Could you use double quotes everywhere ?

File: libraries/classes/Database/Routines.php [hunk 9]
```diff
@@ -1471,65 +1440,48 @@ public function getExecuteForm(array $routine): string
...
-            $retval .= "<input type='hidden' name='ajax_request' value='true'>";
         }
-        $retval .= "</form>\n\n";
-        $retval .= "<!-- END ROUTINE EXECUTE FORM -->\n\n";
 
-        return $retval;
+        return $this->template->render('database/routines/execute_form', [
+            'routine' => $routine,
+            'ajax' => $this->response->isAjax(),
+            'cfg' => $cfg,
```
Comment (williamdes) [inline libraries/classes/Database/Routines.php:1481]:
```suggestion
            'show_function_fields' => $cfg['ShowFunctionFields'],
```

File: libraries/classes/Database/Routines.php [hunk 10]
```diff
@@ -1471,65 +1440,48 @@ public function getExecuteForm(array $routine): string
...
+            'noSupportTypes' => Util::unsupportedDatatypes(),
```
Comment (williamdes) [inline libraries/classes/Database/Routines.php:1482]: seems not used ?

**Comment by williamdes** [comment]
> @williamdes do you know what I need to do to fix the selenium test issues?

I am not sure you have something to do, you can re-base on the latest master branch that is fixed.
You can find some help on our [wiki](https://github.com/phpmyadmin/phpmyadmin/wiki/GIT_MANUAL)

**Comment by mauriciofauth** [comment]
I did some minor fixes to your pull request and merged it.
Thank you for your contribution, @britbarn.

**Comment by britbarn** [comment]
> I did some minor fixes to your pull request and merged it.
> Thank you for your contribution, @britbarn.

Thank you! I'm glad to contribute, and I intend on doing so again.

**Pull request merged**